core/blockchain_reader.go

```diff
@@ -277,12 +277,12 @@ func (bc *BlockChain) GetTransactionLookup(hash common.Hash) (*rawdb.LegacyTxLoo
 	if tx == nil {
 		progress, err := bc.TxIndexProgress()
 		if err != nil {
-			return nil, nil, nil
+			return nil, nil, err
```

Comment:
Error is only returned if (a) transaction is not found (b) and background indexing is still going on.

It's an internal system error here, so please revert this case.